torch/jit/__init__.py

```diff
@@ -712,18 +713,24 @@ def _try_compile_weak_script(fn):
         return entry["compiled_fn"]
 
 
-def script(fn, optimize=True, _frames_up=0, _rcb=None):
+def script(obj, optimize=True, _frames_up=0, _rcb=None):
     if not _enabled:
-        return fn
+        return obj
     if _rcb is None:
         _rcb = _jit_internal.createResolutionCallback(_frames_up + 1)
-    ast = get_jit_ast(fn, is_method=False)
-    mod = ScriptModule()
-    _jit_script_compile(mod, ast, _rcb, get_default_args(fn))
+    if inspect.isclass(obj):
+        mod = ScriptClass(obj.__name__)
```

Comment:
We should consider namespaces as well. These are going to be globally registered in JIT. We may want to use the fully qualified python module name or allow the user to specify qualified name for a type.

Reply:
Let's leave this for a follow-up PR. It's a little tricky, since right now a user may want to annotate a type `Foo` but we have no way of going from that name to a fully-qualified type without pulling a python dependency into the type parser.